sourir.ml

```diff
@@ -85,11 +85,47 @@ let () =
           Disasm.pretty_print_version stderr (v, instrs);
           Scope.explain_incompatible_scope stderr scope1 scope2 pc;
           flush stderr;
-        | _ -> assert(false) end;
+        | _ -> assert(false)
+        end;
         exit 1
       end
     end;
 
+    begin try Check.well_formed program with
+    | Check.MissingMain ->
+      Printf.eprintf "Program is missing an explicit or implicit main function\n";
+      exit 1
+    | Check.DuplicateFunctionDeclaration f ->
+      Printf.eprintf "Duplicate function declaration %s\n" f;
+      exit 1
+    | Check.InvalidFunctionDeclaration f ->
```

Comment:
This error (and message) isn't very helpful. Why is the function invalid?

Currently, it looks like this is only raised if the main function has a non-empty parameter list. Will this exception be used in other cases?